### PR TITLE
DGS-5490: Show `Metadata` and `Ruleset` in schema describe

### PR DESCRIPTION
**Pull request by xiaoyali97**
Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----
Fix bug in https://github.com/confluentinc/cli/pull/1837 to allow optional `Metadata` and `Ruleset` when registering schema and updating config. 

References
----------

Test & Review
-------------
Added some integration tests. 
Also did manual test:
- Essential package: 
```
# create schema without ruleset success
xli@Xiaoya-Li-MBP16 ~/git/go/src/github.com/confluentinc/cli
 % ./dist/confluent_darwin_amd64_v1/confluent schema-registry schema create --subject payments --schema ./test/fixtures/input/schema-registry/schema-example.json --metadata ./test/fixtures/input/schema-registry/schema-metadata.json
Enter your Schema Registry API key: M44GMXNDEWXJ6X4Y
Enter your Schema Registry API secret: ****************************************************************
Successfully registered schema with ID 100001

# create schema with ruleset fails
xli@Xiaoya-Li-MBP16 ~/git/go/src/github.com/confluentinc/cli
 % ./dist/confluent_darwin_amd64_v1/confluent schema-registry schema create --subject payments --schema ./test/fixtures/input/schema-registry/schema-example.json --metadata ./test/fixtures/input/schema-registry/schema-metadata.json --ruleset ./test/fixtures/input/schema-registry/schema-ruleset.json
Error: {"error_code":40302,"message":"Upgrade to Stream Governance Advanced package to use schema rules"}
```

- Advanced package:
```
# create schema with ruleset success
xli@Xiaoya-Li-MBP16 ~/git/go/src/github.com/confluentinc/cli
 % ./dist/confluent_darwin_amd64_v1/confluent schema-registry schema create --subject payments --schema ./test/fixtures/input/schema-registry/schema-example.json --metadata ./test/fixtures/input/schema-registry/schema-metadata.json --ruleset ./test/fixtures/input/schema-registry/schema-ruleset.json
Enter your Schema Registry API key: H3XGVBJRA4O7UDPN
Enter your Schema Registry API secret: ****************************************************************
Successfully registered schema with ID 100085

# describe
xli@Xiaoya-Li-MBP16 ~/git/go/src/github.com/confluentinc/cli
 % ./dist/confluent_darwin_amd64_v1/confluent schema-registry schema describe 100085
Schema ID: 100085
Type: 
Schema:
{"type":"record","name":"Employee","namespace":"Example","fields":[{"name":"Name","type":"string"},{"name":"Age","type":"int"}]}
Metadata:
{
  "properties": {
    "email": "bob@acme.com",
    "owner": "Bob Jones"
  }
}
Ruleset:
{
  "domainRules": [
    {
      "name": "checkSsnLen",
      "kind": "CONDITION",
      "mode": "WRITE",
      "type": "CEL",
      "expr": "size(message.ssn) == 9"
    }
  ]
}

```

Open Questions / Follow-ups
---------------------------

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,7 +6,6 @@ require (
 	github.com/antihax/optional v1.0.0
 	github.com/aws/aws-sdk-go v1.44.262
 	github.com/billgraziano/dpapi v0.4.0
-	github.com/brianstrauch/cobra-shell v0.4.0
```
Comment (brianstrauch) [inline go.mod:9]: Run `go mod tidy`!

File: internal/cmd/schema-registry/command_cluster_update.go [hunk 2]
```diff
@@ -102,35 +102,63 @@ func (c *command) getConfigUpdateRequest(cmd *cobra.Command) (srsdk.ConfigUpdate
...
-	if err := readPathFlag(cmd, "metadata-defaults", &metadataDefaults); err != nil {
-		return updateReq, err
+	updateReq = srsdk.ConfigUpdateRequest{
+		Compatibility:      strings.ToUpper(compatibility),
+		CompatibilityGroup: compatibilityGroup,
 	}
 
-	if err := readPathFlag(cmd, "metadata-overrides", &metadataOverrides); err != nil {
-		return updateReq, err
+	metadataDefaultsPath, err := cmd.Flags().GetString("metadata-defaults")
```
Comment (brianstrauch) [inline internal/cmd/schema-registry/command_cluster_update.go:110]:
```suggestion
	metadataDefaults, err := cmd.Flags().GetString("metadata-defaults")
```

File: internal/cmd/schema-registry/command_cluster_update.go [hunk 3]
```diff
@@ -102,35 +102,63 @@ func (c *command) getConfigUpdateRequest(cmd *cobra.Command) (srsdk.ConfigUpdate
...
+	if err != nil {
+		return srsdk.ConfigUpdateRequest{}, err
+	}
+	if metadataDefaultsPath != "" {
+		var metadataDefaults srsdk.Metadata
+		err := read(metadataDefaultsPath, &metadataDefaults)
+		if err != nil {
+			return srsdk.ConfigUpdateRequest{}, err
+		}
+		updateReq.DefaultMetadata = &metadataDefaults
```
Comment (brianstrauch) [inline internal/cmd/schema-registry/command_cluster_update.go:120]:
```suggestion
		err := read(metadataDefaultsPath, updateReq.DefaultMetadata)
		if err != nil {
			return srsdk.ConfigUpdateRequest{}, err
		}
```

File: internal/cmd/schema-registry/command_schema_describe.go [hunk 4]
```diff
@@ -112,7 +112,7 @@ func describeById(id string, srClient *srsdk.APIClient, ctx context.Context) err
 		return err
 	}
 
-	return printSchema(schemaID, schemaString.Schema, schemaString.SchemaType, schemaString.References)
+	return printSchema(schemaID, schemaString.Schema, schemaString.SchemaType, schemaString.References, schemaString.Metadata, schemaString.RuleSet)
```
Comment (brianstrauch) [inline internal/cmd/schema-registry/command_schema_describe.go:115]: Can we just pass `schemaString`? (Also, it looks like `schemaString` is a `struct`, not a `string`)

File: test/fixtures/output/schema-registry/config/describe-global.golden [hunk 5]
```diff
@@ -1,7 +1,9 @@
-+---------------------+------+
-| Compatibility Level | FULL |
-| Metadata Defaults   | {}   |
-| Metadata Overrides  | {}   |
-| Ruleset Defaults    | {}   |
-| Ruleset Overrides   | {}   |
-+---------------------+------+
++---------------------+------------------------------+
```
Comment (brianstrauch) [inline test/fixtures/output/schema-registry/config/describe-global.golden:1]: This output looks much nicer!

File: internal/cmd/schema-registry/command_cluster_update.go (hunk 3 above)
Comment (brianstrauch) [reply internal/cmd/schema-registry/command_cluster_update.go:120]: I think you would need to add `updateReq.DefaultMetadata = new(srsdk.Metadata)` on the line before!

Comment (brianstrauch) [reply internal/cmd/schema-registry/command_cluster_update.go:120]: Either way is fine, but this way doesn't create a new variable.

File: internal/cmd/schema-registry/command_cluster_update.go [hunk 6]
```diff
@@ -102,35 +102,59 @@ func (c *command) getConfigUpdateRequest(cmd *cobra.Command) (srsdk.ConfigUpdate
...
 	}
 
-	var metadataDefaults srsdk.Metadata
-	var metadataOverrides srsdk.Metadata
-	var rulesetDefaults srsdk.RuleSet
-	var rulesetOverrides srsdk.RuleSet
-
-	if err := readPathFlag(cmd, "metadata-defaults", &metadataDefaults); err != nil {
-		return updateReq, err
+	updateReq = srsdk.ConfigUpdateRequest{
```
Comment (brianstrauch) [inline internal/cmd/schema-registry/command_cluster_update.go:105]: It's more idiomatic to return `srsdk.ConfigUpdateRequest{}, err` above.
```suggestion
	updateReq := srsdk.ConfigUpdateRequest{
```

File: internal/cmd/schema-registry/command_config_describe.go [hunk 7]
```diff
@@ -83,35 +83,45 @@ func describeSchemaConfig(cmd *cobra.Command, srClient *srsdk.APIClient, ctx con
 		}
 	}
 
-	defaultMetadata, err := json.Marshal(config.DefaultMetadata)
-	if err != nil {
-		return err
+	configOut := &configOut{
```
Comment (brianstrauch) [inline internal/cmd/schema-registry/command_config_describe.go:86]:
```suggestion
	out := &configOut{
```

File: internal/cmd/schema-registry/command_schema_describe.go (hunk 4 above)
Comment (brianstrauch) [reply internal/cmd/schema-registry/command_schema_describe.go:115]: Can we nest one struct in the other? This seems very weird to me.

Comment (xiaoyali97) [reply internal/cmd/schema-registry/command_schema_describe.go:115]: yeah unfortunately that's returned by the API 
https://github.com/confluentinc/schema-registry-sdk-go/blob/master/api_default.go#L373
vs 
https://github.com/confluentinc/schema-registry-sdk-go/blob/master/api_default.go#L385

Comment (brianstrauch) [reply internal/cmd/schema-registry/command_schema_describe.go:115]: Confusing how `SchemaString` is not a string... Ok. We can leave this as-is.

**Review by brianstrauch — APPROVED** [review]
Thank you!

**Pull request merged**